include/irrUString.h

```diff
@@ -1323,7 +1323,7 @@ class ustring16
 
 		// Determine if the string is long enough for a BOM.
 		u32 len = 0;
-		const uchar8_t* p = other;
+		const u8* p = (const u8*)other;
```

Comment:
should be
```cpp
		const u8* p = reinterpret_cast<const u8*>(other);
```

Reply:
Yes, using reinterpret_cast also works.

The source file uses a mix of C-style casts and C++ reinterpret_cast, so I don't see that there should be a preference.

Reply:
I pushed a version using reinterpret_cast. Tested with GCC 9.4.0 on Linux.